### PR TITLE
upgrade to nextflow 24.04.2

### LINKED ISSUE #34: Upgrade to last version of Nextflow
evaluate if we can upgrade to NF 24.x

### PR DESCRIPTION
**Pull request by jagedn**
- upgrade gradle to required version
- upgrade required nf to version 24.04.2

closes #34

### CODE REVIEW

**Review by matthdsm — APPROVED** [review]

File: plugins/nf-nomad/src/test/nextflow/nomad/NomadDSLSpec.groovy [hunk 1]
```diff
@@ -57,8 +60,13 @@ class NomadDSLSpec  extends Dsl2Spec{
...
                     @Override
+                    protected Manifest readManifestFromDirectory(Path pluginPath) {
+                        def manifestPath= getManifestPath(pluginPath)
+                        final input = Files.newInputStream(manifestPath)
+                        return new Manifest(input)
+                    }
                     protected Path getManifestPath(Path pluginPath) {
-                        return pluginPath.resolve('build/resources/main/META-INF/MANIFEST.MF')
+                        return pluginPath.resolve('build/tmp/jar/MANIFEST.MF')
                     }
```
Comment (abhi18av) [inline plugins/nf-nomad/src/test/nextflow/nomad/NomadDSLSpec.groovy:70]: @jagedn , I'm just curious about this one - was this change made in upstream Nextflow as well?

File: plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadScriptLauncher.groovy [hunk 2]
```diff
@@ -31,7 +31,7 @@ import nextflow.util.Escape
 class NomadScriptLauncher extends BashWrapperBuilder {
 
     NomadScriptLauncher(TaskBean taskBean) {
-        super(taskBean, new SimpleFileCopyStrategy(taskBean))
```
Comment (abhi18av) [inline plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadScriptLauncher.groovy:34]: I think in future we might have to put this back in but for now, let's focus on the `dev/prod` infras we have with @tomiles

**Review by abhi18av — APPROVED** [review]
@jagedn , thanks for the PR this looks good to merge. 

I just have added some comments to confirm with you.

File: plugins/nf-nomad/src/test/nextflow/nomad/NomadDSLSpec.groovy (hunk 1 above)
Comment (jagedn) [reply plugins/nf-nomad/src/test/nextflow/nomad/NomadDSLSpec.groovy:70]: yes, I think was due to upgrade gradle to 8.3

Comment (abhi18av) [reply plugins/nf-nomad/src/test/nextflow/nomad/NomadDSLSpec.groovy:70]: Ah, okay - then it makes sense 👍

File: plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadScriptLauncher.groovy (hunk 2 above)
Comment (abhi18av) [reply plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadScriptLauncher.groovy:34]: Okay, then happy to go ahead with the removal as of now 👍

**Pull request merged**